File: testdata/go/level-09-scope-leak-intrafile/src/handlers.go
package demo

type Response struct {
	Code int
}

func (r *Response) String() string {
	return "response"
}

func HandleA() string {
	r := &Response{Code: 200}
	return r.String()
}

func HandleB() string {
	r := &Response{Code: 404}
	return r.String()
}

File: testdata/rust/level-09-scope-leak-intrafile/src/handlers.rs
struct Response {
    code: i32,
}

impl Response {
    fn new(code: i32) -> Response {
        Response { code }
    }
}

fn handle_a() -> Response {
    Response::new(200)
}

fn handle_b() -> Response {
    Response::new(404)
}
